docs/design.md

```diff
@@ -0,0 +1,785 @@
+does not provide an official media type for YAML, this API will use
+`application/x-yaml`.
+
+### POST `/documents`
+
+Accepts a multi-document YAML body and creates a new revision which adds
+those documents. Updates are detected based on exact match to an existing
+document of `schema` + `metadata.name`. Documents are "deleted" by including
+documents with the tombstone metadata schema, such as:
+
+```yaml
+---
+schema: any-namespace/AnyKind/v1
+metadata:
+  schema: metadata/Tombstone/v1
+  name: name-to-delete
+...
+```
+
+This endpoint is the only way to add, update, and delete documents. This
+triggers Deckhand's internal schema validations for all documents.
+
+If no changes are detected, a new revision should not be created. This allows
+services to periodically re-register their schemas without creating
+unnecessary revisions.
+
+### GET `/revisions/{revision_id}/documents`
+
+Returns a multi-document YAML response containing all the documents matching
+the filters specified via query string parameters. Returned documents will be
+as originally posted with no substitutions or layering applied.
+
+Supported query string parameters:
+
+* `schema` - string, optional - The top-level `schema` field to select. This
+  may be partially specified by section, e.g., `schema=promenade` would select all
+  `kind` and `version` schemas owned by promenade, or `schema=promenade/Node`
+  which would select all versions of `promenade/Node` documents. One may not
+  partially specify the namespace or kind, so `schema=promenade/No` would not
+  select `promenade/Node/v1` documents, and `schema=prom` would not select
+  `promenade` documents.
+* `metadata.name` - string, optional
+* `metadata.layeringDefinition.abstract` - string, optional - Valid values are
+  the "true" and "false".
+* `metadata.layeringDefinition.layer` - string, optional - Only return documents from
+  the specified layer.
+* `metadata.label` - string, optional, repeatable - Uses the format
+  `metadata.label=key=value`. Repeating this parameter indicates all
+  requested labels must apply (AND not OR).
+
+### GET `/revisions/{revision_id}/rendered-documents`
+
+Returns a multi-document YAML of fully layered and substituted documents. No
+abstract documents will be returned. This is the primary endpoint that
+consumers will interact with for their configuration.
+
+Valid query parameters are the same as for
+`/revisions/{revision_id}/documents`, minus the paremters in
+`metadata.layeringDetinition`, which are not supported.
+
+### GET `/revisions`
+
+Lists existing revisions and reports basic details including a summary of
+validation status for each `deckhand/ValidationPolicy` that is part of that
+revision.
+
+Sample response:
+
+```yaml
+---
+count: 7
+next: https://deckhand/revisions?limit=2&offset=2
+prev: null
+results:
+  - id: 0
+    url: https://deckhand/revisions/0
+    createdAt: 2017-07-14T21:23Z
+    validationPolicies:
+      site-deploy-validation:
```

Comment:
Is `site-deploy-validation` an example of a particular validation or is it the overall status of the entire set of validations? If not, why not include just the overall status of the set of validations for each revision?